azurerm/internal/services/compute/validation.go

```diff
@@ -168,9 +168,13 @@ func validateDiskSizeGB(v interface{}, _ string) (warnings []string, errors []er
 
 func validateManagedDiskSizeGB(v interface{}, _ string) (warnings []string, errors []error) {
 	value := v.(int)
-	if value < 0 || value > 32767 {
+	if value < 0 || value > 65536 {
 		errors = append(errors, fmt.Errorf(
-			"The `disk_size_gb` can only be between 0 and 32767"))
+			"The `disk_size_gb` can only be between 0 and 65536"))
+	}
+	if value > 32767 {
+		warnings = append(warnings,
+			"The `disk_size_gb` can be higher than 32767 only for `storage_account_type` UltraSSDLRS")
 	}
```

Comment:
whilst exposed in the SDK, unfortunately warnings don't work at this time - so could we remove this?

Reply:
For sure. Thanks for you feedback.